### ISSUE TITLE
Allows access to the pairwise distances computed during training

### ISSUE THREAD

**Issue by ebursztein** [tensorflow/similarity#36]
Ensure callback can access the distances computed during the loss computation. Potentially store them in the loss object by subclassing the loss wrapper?
